src/librustc_infer/infer/mod.rs

```diff
@@ -1034,7 +1034,7 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
 
         Some(self.commit_if_ok(|snapshot| {
             let (ty::SubtypePredicate { a_is_expected, a, b }, placeholder_map) =
-                self.replace_bound_vars_with_placeholders(predicate);
+                self.replace_bound_vars_with_placeholders(&predicate);
```

Comment:
Somewhat related, I think it might be a good idea to change `Binder` to also require `T: Copy` and to move the reference inwards where this is not the case, i.e. `&Binder(T) -> `Binder(&T)`.